src/views/ActivitiesAll/index.js

```diff
@@ -102,22 +166,23 @@ export default class GordonActivitiesAll extends Component {
                 sessions,
                 activities: newActivities,
                 allActivities: newActivities,
-                loading: false,
+                types,
               });
+              break;
             }
```

Comment:
I know they are not exactly the same, but could the sections on lines 85-123 and lines 153-185 be replaced with a single function call?

Reply:
Yep, I cleaned it up